KBaseSearchEngine.spec

```diff
@@ -261,6 +261,8 @@ module KBaseSearchEngine {
           pagination and sorting.
       total - total number of found objects.
       search_time - common time in milliseconds spent.
+      list<ObjectData> objects - Result of search. Deleted or inaccessible objects are removed. List, before removal of
```

Comment:
I'm having a hard time understanding the second sentence. Also, the explanation should include the eventually consistent angle - this sounds like there's something really stupid about the design.

Reply:
How about: 
list<ObjectData> objects - Result of search. Object in transient inaccessible states, such as undergoing deletion or permission change may be included in search total. These inaccessible objects are removed after page is filled.

Reply:
I think that'd still be pretty confusing to someone reading the spec for the first time. Maybe something like 

```
Results of search. In rare cases, due to time delay in updating the search index from changes in the source data, objects returned from the search index may not be accessible at the data source. The API will remove these objects from the returned search results which means the returned results may have fewer objects than a full page.
```

What do you think?